lib/rules/no-unused-prop-types.js

```diff
@@ -22,7 +22,15 @@ const docsUrl = require('../util/docsUrl');
 const DIRECT_PROPS_REGEX = /^props\s*(\.|\[)/;
 const DIRECT_NEXT_PROPS_REGEX = /^nextProps\s*(\.|\[)/;
 const DIRECT_PREV_PROPS_REGEX = /^prevProps\s*(\.|\[)/;
-const LIFE_CYCLE_METHODS = ['componentWillReceiveProps', 'shouldComponentUpdate', 'componentWillUpdate', 'componentDidUpdate'];
+const LIFE_CYCLE_METHODS = [
+  'componentDidUpdate',
+  'componentWillReceiveProps',
+  'componentWillUpdate',
+  'getDerivedStateFromProps',
```

Comment:
these new methods should only be checked if the React version pragma from settings is >= 16.3.

Reply:
Ah, yes. Thanks for pointing that out.